modules/tinymce/src/core/main/ts/delete/MergeBlocks.ts

```diff
@@ -55,10 +55,30 @@ const nestedBlockMerge = (
 
 const sidelongBlockMerge = (rootNode: SugarElement<Node>, fromBlock: SugarElement<Element>, toBlock: SugarElement<Element>): Optional<CaretPosition> => {
   if (Empty.isEmpty(toBlock)) {
-    Remove.remove(toBlock);
     if (Empty.isEmpty(fromBlock)) {
-      PaddingBr.fillWithPaddingBr(fromBlock);
+      const getInlineToBlockDescendants = (el: SugarElement<Element>) => {
+        const helper = (node: SugarElement<Element>, elements: SugarElement<Element>[]): SugarElement<Element>[] =>
+          Traverse.firstChild(node).fold(
+            () => elements,
+            (child) => ElementType.isInline(child) ? helper(child, elements.concat(Replication.shallow(child))) : elements
+          );
+        return helper(el, []);
+      };
+
+      const newFromBlockDescendants = Arr.foldr(
+        getInlineToBlockDescendants(toBlock),
+        (element: SugarElement<Element>, wrapper) => {
+          Insert.wrap(element, wrapper);
+          return wrapper;
```

Comment:
To make this clearer, as per my other comments, I suggest
```suggestion
        (element: SugarElement<Element>, descendant) => {
          Insert.wrap(element, descendant);
          return descendant;
```